WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -274,34 +231,38 @@ private void cancelUpload() {
     }
 
     private void dispatchUploadAction(@NonNull final MediaModel media) {
-        AppLog.i(T.MEDIA, "Dispatching upload action for media with local id: " + media.getId() +
+        AppLog.i(AppLog.T.MEDIA, "Dispatching upload action for media with local id: " + media.getId() +
                 " and path: " + media.getFilePath());
         media.setUploadState(UploadState.UPLOADING.name());
         mDispatcher.dispatch(MediaActionBuilder.newUpdateMediaAction(media));
 
         MediaPayload payload = new MediaPayload(mSite, media);
         mDispatcher.dispatch(MediaActionBuilder.newUploadMediaAction(payload));
-
-        if (mListener != null) {
-            mListener.onUploadBegin(mCurrentUpload);
-        }
     }
 
     private void dispatchCancelAction(@NonNull final MediaModel media) {
-        AppLog.i(T.MEDIA, "Dispatching cancel upload action for media with local id: " + media.getId() +
+        AppLog.i(AppLog.T.MEDIA, "Dispatching cancel upload action for media with local id: " + media.getId() +
                 " and path: " + media.getFilePath());
         MediaPayload payload = new MediaPayload(mSite, mCurrentUpload);
         mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
     }
 
+    private void completed(){
+        AppLog.i(AppLog.T.MEDIA, "Media Upload Service > completed");
+        if (mQueue == null || mQueue.size() == 0) {
```

Comment:
Actually this check seems redundant overall - we only ever call `completed()`  if we have no site (impossible to upload) or if we've already checked that there aren't any more items to upload.

What do you think of updating/adding a specific log message in those three cases and just replace `completed()` with a call to `stopSelf()`?